### PR TITLE
Add Linux kernel tests derivation (lkp-tests)

### PR DESCRIPTION
**Pull request by juliuskoskela**
## Description of changes

Add a derivation for Linux Kernel Performance tests (lkp-tests).

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [ ] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status

### CODE REVIEW

File: packages/lkp-tests/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  nativeBuildInputs = [pkgs.makeWrapper];
+  buildInputs = [pkgs.ruby_3_3];
+
+  phases = ["unpackPhase" "installPhase"];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r . $out
+    make install TARGET_DIR_BIN=$out
+  '';
```
Comment (Mic92) [inline packages/lkp-tests/default.nix:24]: Enables parallel builds. Maybe check if it doesn't break the build. In most projects it's fine:

```suggestion
  enableParallelBuilding = true;
  preInstall = ''
    mkdir -p $out/bin
    cp -r . $out
  '';
  installFlags = ["TARGET_DIR_BIN=$(out)"];
```

File: packages/lkp-tests/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+{pkgs}:
+pkgs.stdenv.mkDerivation {
```
Comment (Mic92) [inline packages/lkp-tests/default.nix:5]:
```suggestion
{pkgs}:
pkgs.stdenv.mkDerivation {
```
```suggestion
{pkgs}:
pkgs.stdenv.mkDerivation {
```
```suggestion
{ stdenv, fetchFromGitHub, makeWrapper, ruby_3_3, lib }:
pkgs.stdenv.mkDerivation {
```

File: packages/lkp-tests/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = pkgs.fetchFromGitHub {
+    owner = "intel";
+    repo = "lkp-tests";
+    rev = "master";
+    sha256 = "sha256-owXTb5+O/V7Fu4myakx2h5wBph4LCLOW/N5Qq8D2Qec=";
+  };
+
+  nativeBuildInputs = [pkgs.makeWrapper];
+  buildInputs = [pkgs.ruby_3_3];
```
Comment (Mic92) [inline packages/lkp-tests/default.nix:16]:
```suggestion
  nativeBuildInputs = [makeWrapper];
  buildInputs = [ruby_3_3];
```

File: packages/lkp-tests/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  phases = ["unpackPhase" "installPhase"];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r . $out
+    make install TARGET_DIR_BIN=$out
+  '';
+
+  meta = with pkgs.lib; {
```
Comment (Mic92) [inline packages/lkp-tests/default.nix:26]:
```suggestion
  meta = with lib; {
```

**Comment by juliuskoskela** [comment]
Hey @Mic92 thank you for your review comments! Unfortunately this PR is not going to go through. I had misunderstood how lkp-tests actually works. It's basically a "package manager" for tests and thus not compatible with NixOS as is. It i[nstalls tests using Arch Linux PKGBUILD definitions](https://github.com/intel/lkp-tests/tree/master/programs) and then [parameterizes those tests into jobs using yaml templates](https://github.com/intel/lkp-tests/tree/master/jobs). This is all driven with bash and ruby scripts that are locked to [supported distros](https://github.com/intel/lkp-tests/tree/master/distro).

I made a little test framework with Nix on how we could derive the tests in lkp-tests for Nix, which can be found [here](https://github.com/juliuskoskela/nix-bench). I derived [one example tests from lkp-tests ](https://github.com/juliuskoskela/nix-bench/blob/main/pkgs/adrestia.nix) and [parameterized it](https://github.com/juliuskoskela/nix-bench/blob/main/tests/adrestia/default.nix) using a couple of library [function](https://github.com/juliuskoskela/nix-bench/blob/main/lib/default.nix) I created.

**Pull request closed without merging**